File: generator-backend/src/main/java/de/th/bingen/master/backend/model/request/Request.java
package de.th.bingen.master.backend.model.request;

import de.th.bingen.master.backend.model.request.container.Container;
import de.th.bingen.master.backend.model.request.service.Service;

import javax.validation.Valid;
import java.util.ArrayList;
import java.util.List;

public class Request {
    @Valid
    private List<Container> containers = new ArrayList<>();
    @Valid
    private List<Database> databases = new ArrayList<>();
    @Valid
    private List<Service> services = new ArrayList<>();
    @Valid
    private List<Link> links = new ArrayList<>();
    private String templateName;

    public String getTemplateName() {
        return templateName;
    }

    public void setTemplateName(String templateName) {
        this.templateName = templateName;
    }

    public List<Container> getContainers() {
        return containers;
    }

    public void setContainers(List<Container> containers) {
        this.containers = containers;
    }

    public List<Database> getDatabases() {
        return databases;
    }

    public void setDatabases(List<Database> databases) {
        this.databases = databases;
    }

    public List<Service> getServices() {
        return services;
    }

    public void setServices(List<Service> services) {
        this.services = services;
    }

    public List<Link> getLinks() {
        return links;
    }

    public void setLinks(List<Link> links) {
        this.links = links;
    }

    public void addContainer(Container container) {
        containers.add(container);
    }

    public void addService(Service service) {
        services.add(service);
    }

    public void addAllLinks(ArrayList<Link> links) {
        this.links.addAll(links);
    }
}

File: generator-frontend/src/app/model/Request.ts
import { Container } from './container/Container';
import { Service } from './service/Service';
import { Link } from './Link';
import { Database } from './database/Database';
import { dataType } from 'angular-http-deserializer/decorators';

/**
 * Request which will get sent to the backend
 */
export class Request {
    /** Name of the template */
    templateName: string;

    /** Array of requested containers */
    @dataType(Container, true)
    containers: Container[] = [];

    /** Array of requested databases */
    @dataType(Database, true)
    databases: Database[] = [];

    /** Array of requested services */
    @dataType(Service, true)
    services: Service[] = [];

    /** Array of requested links */
    @dataType(Link, true)
    links: Link[] = [];
}